From 1cea998f4e05e9a774e1c5cc423c1501ed0eb411 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Tue, 19 Jun 2018 16:09:26 +0300
Subject: [PATCH 3986/5242] MLK-18636 Input: egalax_ts: Restore call to
 i2c_set_clientdata

commit  b6f37f7179f765f59281977db9099f6052cf095e from
https://source.codeaurora.org/external/imx/linux-imx.git

Upstream dropped the call to i2c_set_client_data as "unneeded" in commit
8300445cc7c1 ("Input: touchscreen - drop calls to platform_set_drvdata and i2c_set_clientdata")

The client_data pointer is used on suspend/resume by downstream commit
76a621ae711b ("MLK-17779 input: egalax_ts: free irq resource before request the line as GPIO")

This causes suspend/resume to crash (and apparently hang) on
imx6qdl-sabresd with LVDS display connected. Fix by adding back the
i2c_set_clientdata call.

Fixes: 76a621ae711b ("MLK-17779 input: egalax_ts: free irq resource before request the line as GPIO")

This could be squashed into MLK-17779

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Acked-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/input/touchscreen/egalax_ts.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/input/touchscreen/egalax_ts.c b/drivers/input/touchscreen/egalax_ts.c
index 9be12f0..94fbd15 100644
--- a/drivers/input/touchscreen/egalax_ts.c
+++ b/drivers/input/touchscreen/egalax_ts.c
@@ -256,6 +256,8 @@ static int egalax_ts_probe(struct i2c_client *client,
 	if (error)
 		return error;
 
+	i2c_set_clientdata(client, ts);
+
 	return 0;
 }
 
-- 
1.7.9.5

